include/nbl/video/decl/IBackendObject.h

```diff
@@ -0,0 +1,38 @@
+#ifndef __NBL_VIDEO_DECLARATION_I_BACKEND_OBJECT_H_INCLUDED__
+#define __NBL_VIDEO_DECLARATION_I_BACKEND_OBJECT_H_INCLUDED__
+
+
+#include "nbl/video/EApiType.h"
+
+#include <type_traits>
+
+namespace nbl::video
+{
+
+class ILogicalDevice;
+
+class IBackendObject
+{
+    public:
+        const ILogicalDevice* m_originDevice2;
+        IBackendObject(const ILogicalDevice* device) : m_originDevice2(device) {}
+
+        IBackendObject(core::smart_refctd_ptr<const ILogicalDevice>&& device);
+
+        E_API_TYPE getAPIType() const;
+
+        bool isCompatibleDevicewise(const IBackendObject* other) const;
+
+        bool wasCreatedBy(const ILogicalDevice* device) const;
```

Comment:
@achalpandeyy  this rolled into a templated casting function (ILogicalDevice to a concrete C{API}LogicalDevice) that returns null on failure would be the ultimate utility

Reply:
@Erfan-Ahmadi TODO